rllib/policy/tf_policy.py

```diff
@@ -245,22 +248,30 @@ def __init__(self,
             tf1.placeholder_with_default(
                 tf.zeros((), dtype=tf.int64), (), name="timestep")
 
+        self._optimizers: List[LocalOptimizer] = []
+        # Backward compatibility and for some code shared with tf-eager Policy.
         self._optimizer = None
-        self._grads_and_vars = None
-        self._grads = None
+
+        self._grads_and_vars: Union[ModelGradients, List[ModelGradients]] = []
+        self._grads: Union[ModelGradients, List[ModelGradients]] = []
         # Policy tf-variables (weights), whose values to get/set via
         # get_weights/set_weights.
         self._variables = None
-        # Local optimizer's tf-variables (e.g. state vars for Adam).
+        # Local optimizer(s)' tf-variables (e.g. state vars for Adam).
         # Will be stored alongside `self._variables` when checkpointing.
-        self._optimizer_variables = None
+        self._optimizer_variables: \
+            Optional[ray.experimental.tf_utils.TensorFlowVariables] = None
 
-        # The loss tf-op.
+        # The loss tf-op(s). Number of losses must match number of optimizers.
+        self._losses = []
+        # Backward compatibility (in case custom child TFPolicies access this
+        # property).
         self._loss = None
```

Comment:
do we have to keep self._loss here, instead of moving everything to use self._losses, because of backward compatibility reasons?
like, someone may have been overriding things with their custom tf policy class?

I thought this is DeveloperAPI, so we should be able to move everything to self._optimizers and self._losses?

Reply:
Correct, lots of users probably still directly access `self._loss` or `self._optimizer` in their TFPolicy child classes.